src/index.js

```diff
@@ -163,6 +164,8 @@ export default function onClickOutsideHOC(WrappedComponent, config) {
 
       handlersMap[this._uid] = event => {
         if (this.componentNode === null) return;
+        
+        if (this.initTimeStamp > event.timeStamp) return;
```

Comment:
Hmm, this feels like it's fixing a race condition. This function should not be able to trigger before the constructor runs, so while this might fix the symptom it's probably better to track down the underlying cause. Thanks to it being a long weekend here in Canada, I should hopefully have time today to dig into this a little.

Reply:
Was there any better solution discussed for this?

Reply:
I guess the counter question would be that modern React with hooks kind of... [doesn't need this hoc anymore](https://github.com/Pomax/react-onclickoutside?tab=readme-ov-file#functional-component-with-usestate-hook), does anyone still _need_ this fixed?

Reply:
We are working on an oldcodebase where its not feasible to migrate to hooks easily, So it would be great if we can fix a solution for the issue.

Reply:
Hooks are over 5 years old at this point, I would _strongly_ encourage you to consider getting on board with modern Reactg if you want to keep using it (understanding full well that's almost certainly not your call to make).

I'll merge this in, but given React's trajectory, this will be a bandaid at best.

Reply:
Also, if this HoC is _that_ important to your codebase, consider taking the first paragraph of the README to heart and giving back any way you can.